### DISCUSSION TITLE
How to use the `opts` keys in `setup()`?

### DISCUSSION THREAD

**Discussion by runiq** [folke/noice.nvim#335] [Q&A]
I want to use the nvim-notify infrastructure for LSP progress messages. The `notify` view is almost perfect for that, I just have to set `replace = true` to not get spammed with LSP progress messages. I though I could do that by overriding the global `view.notify` option like this:

```lua
require("noice").setup({
  lsp = {
    progress = {
      view = "notify",
      opts = {
        -- Override this single option, only for lsp progress messages
        replace = true,
      },
    },
  },
})
```
However, that doesn't appear to work. Instead, I'm currently defining a completely new view like this:

```lua
require("noice").setup({
  lsp = {
    progress = {
      -- Create a completely new view
      view = "lsp_notify",
    },
  },
  views = {
    lsp_notify = {
      backend = "notify",
      fallback = "mini",
      format = "notify",
      replace = true,
      merge = false,
    },
  },
})
```

However, now I have to set all view options again explicitly.

Is what I'm currently doing the correct way, or can I use `lsp.progress.opts` for this somehow?

**Comment by agoodshort** [comment]
Thanks for sharing your new view config. I was looking to setup the LSP messages the exact same way as you.
Have you find the correct way to achieve this since?
